### PR TITLE
Replace mate with external-dns

### PR DESCRIPTION
**Pull request by Raffo**
Initial PR for replacing `mate` with `external-dns`, still needs some testing before merging.

### CODE REVIEW

**Comment by hjacobs** [comment]
Latest Docker image is now: registry.opensource.zalan.do/teapot/external-dns:v0.3.0-alpha.2

External DNS tag: https://github.com/kubernetes-incubator/external-dns/releases/tag/v0.3.0-alpha.2

**Comment by Raffo** [comment]
Still TODO: properly plan the rollout and make sure there's no problem while switching from mate to external-dns.

File: cluster/manifests/external-dns/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      application: external-dns
+  template:
+    metadata:
+      labels:
+        application: external-dns
+        version: v0.3.0-alpha.2
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        iam.amazonaws.com/role: "{{ .LocalID }}-app-external-dns"
+    spec:
```
Comment (mikkeloscar) [inline cluster/manifests/external-dns/deployment.yaml]: `tolerations:` definition is missing, see mate deployment below.

Comment (Raffo) [reply cluster/manifests/external-dns/deployment.yaml]: Added

**Comment by hjacobs** [comment]
We should not use the cluster ID as the record owner. Our cluster ID contains the AWS account ID (12 digit string) and we should not expose it as `TXT` records.

**Comment by hjacobs** [comment]
Ah, you already changed it :heart:

**Comment by ideahitme** [comment]
this probably requires more discussion, if we are going to roll out mate/external-dns at the same time and have the migration script to ease up the process

**Comment by ideahitme** [comment]
migration script: https://github.bus.zalan.do/teapot/external-dns-migration

**Comment by ideahitme** [comment]
@Raffo @mikkeloscar @hjacobs I re-added Mate IAM Policy document and included the migration script. Only thing pending is the part handling deletion of mate deployment from clusters :)

**Comment by hjacobs** [comment]
@ideahitme you can add the `deletions.yaml` to remove Mate.

**Comment by szuecs** [comment]
https://github.bus.zalan.do/teapot/external-dns-migration should be in the mate repository, such that other mate users can also migrate to external-dns

**Comment by ideahitme** [comment]
@szuecs yup, will consider adding it later on

**Comment by ideahitme** [comment]
just need to fix the versions of external-dns and external-dns-migration, rest is done

**Comment by ideahitme** [comment]
@hjacobs @Raffo @mikkeloscar @szuecs @linki need review

**Comment by Raffo** [comment]
👍

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**